cycler/__init__.py

```diff
@@ -82,7 +82,8 @@ def _process_keys(
     l_key: set[K] = set(l_peek.keys())
     r_key: set[K] = set(r_peek.keys())
     if l_key & r_key:
-        raise ValueError("Can not compose overlapping cycles")
+        raise ValueError(f"Cannot compose overlapping cycles, duplicate key(s): {l_key & r_key}")
```

Comment:
We could walrus this one, I think
```suggestion
    if common_keys := l_key & r_key:
        raise ValueError(f"Cannot compose overlapping cycles, duplicate key(s): {common_keys}")
```

Reply:
Indeed that makes it a bit cleaner.